xbmc/addons/kodi-dev-kit/include/kodi/addon-instance/Game.h

```diff
@@ -1365,9 +1368,15 @@ class ATTR_DLL_LOCAL CInstanceGame : public IAddonInstance
         static_cast<CInstanceGame*>(instance->toAddon->addonInstance)
             ->RCPostRichPresenceUrl(cppUrl, cppPostData, username, token, gameID, richPresence);
     if (!cppUrl.empty())
-      *url = strdup(cppUrl.c_str());
+    {
+      *url = new char[cppUrl.size() + 1];
+      strcpy(*url, cppUrl.c_str());
```

Comment:
We like functions in the C++ namespace (std::strcpy), but it might be more idomatic to use std::copy

```c++
*url = new char[cppUrl.size() + 1];
std::copy(cppUrl.begin(), cppUrl.end(), *url);
```

`new` initializes also, so you _should_ need the following line

```c++
(*url)[cppUrl.length()] = '\0';
```

Reply:
It didn't work without the `'\0'`

Reply:
then I'm wrong, better include \0